### PR TITLE
Fix Issue 18044: std.conv.to for implicitly convertible associative arrays

### PR DESCRIPTION
**Pull request by John-Colvin**

### CODE REVIEW

**Comment by schveiguy** [comment]
LGTM. Can you make a bugzilla entry so it gets in the changelog?

**Comment by wilzbach** [comment]
The issue number needs to be mentioned in the git commit message. You can verify this is the bot lists the issue. I'm currently on the road, so I just removed auto-merge. Sorry

**Comment by John-Colvin** [comment]
@wilzbach is that documented anywhere? I always forget what I'm supposed to do.

**Comment by wilzbach** [comment]
> Your PR doesn't reference any Bugzilla issue.
> If your PR contains non-trivial changes, please reference a Bugzilla issue or create a manual changelog.

Yes, just read the boy's comment.

**Comment by John-Colvin** [comment]
Yes, I know I needed to make a bugzilla entry, I was just being lazy. The bot doesn't tell me anything about *how* and *where* to reference the issue once created.

**Comment by wilzbach** [comment]
> Yes, I know I needed to make a bugzilla entry, I was just being lazy. The bot doesn't tell me anything about how and where to reference the issue once created.

It does. Just follow the referenced link: https://github.com/dlang-bots/dlang-bot#automated-references

> Have I done it the right way now?

Nope. As said once the git message references the issue, the bot will update its comment and you will see a table listing the issue.

**Comment by wilzbach** [comment]
> Nope. As said once the git message references the issue, the bot will update its comment and you will see a table listing the issue.

I think you just need to remove the colon :
The RegExp for detecting the issues is on the Bugzilla side. Nothing we can do about this.

**Comment by John-Colvin** [comment]
Ah, ok, thanks. Sorry, I read the "reference a Bugzilla issue" section in your quote and missed that in the original it had a link.

**Comment by wilzbach** [comment]
I am sorry for the horrible help - I usually could have simply force-pushed the commit message, but unfortunately it's a bit hard without a computer. Glad you managed to win the fight against the bot :)

**Comment by schveiguy** [comment]
> The issue number needs to be mentioned in the git commit message.

Is this a requirement for it to get in the changelog? I knew this wasn't going to auto close, but I was just going to close it manually in this case :)

**Comment by wilzbach** [comment]
> Is this a requirement for it to get in the changelog? I knew this wasn't going to auto close, but I was just going to close it manually in this case :)

Yes, I assume no one read the Readme of the bot or the wiki entry?
tl;dr: the GitHub Bugzilla integration handles closing the issues, the dlang/tools/changed.d script parses the git log between two releases only. Hence, if the issue isn't mentioned in the git log, it won't appear in the log.
BTW since a couple of months it's possible to preview the changelog on DAutoTest: http://dtest.dlang.io/artifact/website-bb685c0e96626967d952e71a0ef838246140b3c1-67c70e66687de80247893d87d6f0b012/web/changelog/2.078.0_pre.html or on https://dlang.org/changelog/2.078.0_pre.html, but issues will only be displayed if they are closed, that's why there are no changes listed on DAutoTest.

**Comment by schveiguy** [comment]
> Yes, I assume no one read the Readme of the bot or the wiki entry?

You assumed correctly :) I am a lazy programmer after all...

Honestly, I didn't realize this had anything to do with git, I just thought the changelog consisted of closed bugs in the time between releases. But the way it is makes a hell of a lot more sense -- some bugs are closed as "WORKSFORME", which have nothing to do with the release.
